robolectric/src/main/java/org/robolectric/internal/dependency/LocalDependencyResolver.java

```diff
@@ -8,9 +8,11 @@
 
 public class LocalDependencyResolver implements DependencyResolver {
   private File offlineJarDir;
+  private final DependencyResolver dependencyResolver;
 
-  public LocalDependencyResolver(File offlineJarDir) {
+  public LocalDependencyResolver(DependencyResolver dependencyResolver, File offlineJarDir) {
```

Comment:
I think it is better to keep this class focused on loading local dependencies rather that having extra fall back logic associated with it. DependencyResolver is an interface, so you could equally create CompositeDependencyResolver class that takes an Iterable<DependencyResolver> trying starting with the first and iterating through the resolvers till it has found one.

The reason is, different projects use different build systems, we for instance have all our dependency jars checked in to source control rather than fetching them at runtime.